agent/ui_endpoint.go

```diff
@@ -339,3 +339,42 @@ func modifySummaryForGatewayService(
 		}
 	}
 }
+
+// GET /v1/internal/ui/gateway-intentions/:gateway
+func (s *HTTPServer) UIGatewayIntentions(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	var args structs.IntentionQueryRequest
+	if done := s.parse(resp, req, &args.Datacenter, &args.QueryOptions); done {
+		return nil, nil
+	}
+
+	var entMeta structs.EnterpriseMeta
+	if err := s.parseEntMetaNoWildcard(req, &entMeta); err != nil {
+		return nil, err
+	}
+
+	// Pull out the service name
+	name := strings.TrimPrefix(req.URL.Path, "/v1/internal/ui/gateway-intentions/")
+	if name == "" {
+		resp.WriteHeader(http.StatusBadRequest)
+		fmt.Fprint(resp, "Missing gateway name")
+		return nil, nil
+	}
+	args.Match = &structs.IntentionQueryMatch{
+		Type: structs.IntentionMatchDestination,
+		Entries: []structs.IntentionMatchEntry{
+			{
+				Namespace: entMeta.NamespaceOrDefault(),
```

Comment:
```suggestion
				Namespace: entMeta.NamespaceOrEmpty(),
```
You'll want to double check with the API side of IntentionMatch, but I think this change is more appropriate.

The rules of thumb for namespaces are roughly:
1. The API should pass explicit namespace values through to RPC.
2. The API should leave unspecified namespace values empty.
3. RPC should decode the namespace of the current ACL Token and use that to populate empty namespace values arriving from the API.
4. The `"default"` should come from the token logic and not from your own.

Reply:
Ah yep. The token's namespace wouldn't have been considered due to defaulting it here.